docs/broker/filtering.md

```diff
@@ -0,0 +1,336 @@
+Trigger. If the expression evaluates to true, the event is delivered to the
+Trigger's subscriber.
+
+The evaluation environment of the expression may include:
+
+*   Standard CloudEvent fields like type and source
+*   Custom extensions dynamically parsed from the event
+*   Data fields parsed from the event, if the data content type is known to be
+    parseable
+
+Since parsing dynamic extension and data fields may be costly, the Trigger must
+explicitly enable it by setting boolean fields adjacent to the expression. See
+[examples](#examples) for syntax.
+
+The expression language chosen is
+[CEL](https://github.com/google/cel-spec/blob/9cdb3682ba04109d2e03d9b048986bae113bf36f/doc/intro.md)
+as it has the best combination of features for this purpose. See
+[alternatives considered](#alternatives-considered) for comparisons to other
+expression languages.
+
+```yaml
+spec:
+  filter:
+    cel:
+      expression: >
+        ce.type == "com.github.pull.create" &&
+        source == "https://github.com/knative/eventing/pulls/123"
+```
+
+### Why CEL?
+
+CEL is the
+[Common Expression Language](https://github.com/google/cel-spec/blob/9cdb3682ba04109d2e03d9b048986bae113bf36f/doc/langdef.md).
+It is designed for expressing security policy and protocols, and has the
+following
+[guiding philosophy](https://github.com/google/cel-spec/blob/9cdb3682ba04109d2e03d9b048986bae113bf36f/README.md):
+
+*   **Small and fast.** CEL is not Turing complete, so it is easy to understand
+    and implement.
+*   **Extensible.** CEL is designed to be embedded and can be extended with new
+    functionality.
+*   **Developer-friendly.** The language spec was based on experience and
+    usability testing from implementing Firebase Rules.
+
+Additional properties of CEL that make it attractive for Trigger Filtering
+include:
+
+*   **Familiar expression syntax.** Filter expressions written in CEL look and
+    behave like conditional expressions written in the most popular languages.
+*   **Expressions can be compiled and cached for efficiency.** We expect users
+    will create many filters that change rarely in comparison to frequency of
+    evaluations. This usage pattern works in favor of an expression language
+    that can be cached in compiled form.
+*   **Expressions are serializable and composable.** These properties are useful
+    in implementing upstream filter propagation. For example, a Broker could
+    advertise a single CEL expression composed of the union of its Triggers'
+    filters, allowing upstream sources to filter events before publishing to the
+    Broker.
+
+CEL is
+[memory-safe, side-effect-free, and terminating](https://github.com/google/cel-spec/blob/master/doc/langdef.md#overview),
+making it safe and secure to embed in a multi-tenant dispatcher process.
+
+## Examples
+
+### Single field exact match
+
+Specified with the SourceAndType field: `yaml spec: filter: sourceAndType: type:
```

Comment:
This should probably be triple back tick.

Reply:
Good catch, this was mangled by the markdown formatter I was using.